### PR TITLE
Disabling the current gRPC LB method of handling fallback in xds plugin.

### PR DESCRIPTION
**Pull request by vishalpowar**
xDS plugin will need seperate policy for fallback and handling the
transition to and from fallback mode.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,208      Total (<)      2,019,390

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,180,036      Total (>)     11,180,031

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 1]
```diff
@@ -1323,6 +1323,8 @@ void XdsLb::UpdateLocked(const grpc_channel_args& args) {
   ProcessChannelArgsLocked(args);
   // If fallback is configured and the RR policy already exists, update
   // it with the new fallback addresses.
+  // Note: We have disable fallback mode in the code, so this will only happen
```
[the remark this answers is not in the document]
Comment (vishalpowar) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: When I say disable in the fallback i mean in the code. like 1400 is deleted.

**Comment by vishalpowar** [comment]
xDS plugin fallback is going to be different than gRPC LB.
Right now I am cleaning up the parts which will change significantly in future.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 2]
```diff
@@ -1646,13 +1647,9 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
     is_backend_from_grpclb_load_balancer = true;
   } else {
-    // If CreateOrUpdateRoundRobinPolicyLocked() is invoked when we haven't
-    // received any serverlist from the balancer, we use the fallback backends
-    // returned by the resolver. Note that the fallback backend list may be
-    // empty, in which case the new round_robin policy will keep the requested
-    // picks pending.
-    GPR_ASSERT(fallback_backend_addresses_ != nullptr);
-    addresses = grpc_lb_addresses_copy(fallback_backend_addresses_);
+    // This should never be invoked if we do not have serverlist_, as fallback
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: If so, we can remove this `if`branch, and add `GPR_ASSERT(serverlist_ != nullptr);` above.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.0%     -16 [None]                                                                               -1.16Ki  -0.0%
  -0.7%    -112 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                -112  -0.7%
      [DEL] -1.83Ki grpc_core::(anonymous namespace)::XdsLb::CreateOrUpdateRoundRobinPolicyLocked() [clo -1.83Ki  [DEL]
     -38.6%     -80 grpc_core::(anonymous namespace)::XdsLb::OnFallbackTimerLocked                           -80 -38.6%

  -0.0%    -128 TOTAL                                                                                -1.27Ki  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (vishalpowar) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: fair point, removed this piece.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,267      Total (<)      2,019,390

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,180,035      Total (<)     11,180,037

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.0%    -136 [None]                                                                               -3.55Ki  -0.0%
  -3.7%    -592 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                -592  -3.7%
      [DEL] -1.83Ki grpc_core::(anonymous namespace)::XdsLb::CreateOrUpdateRoundRobinPolicyLocked() [clo -1.83Ki  [DEL]
     -38.6%     -80 grpc_core::(anonymous namespace)::XdsLb::OnFallbackTimerLocked                           -80 -38.6%
     -19.1%     -48 grpc_core::(anonymous namespace)::XdsLb::UpdateLocked                                    -48 -19.1%
      -2.8%     -11 [Unmapped]                                                                               -11  -2.8%

  -0.0%    -728 TOTAL                                                                                -4.12Ki  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,040      Total (<)      2,019,390

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,738      Total (<)     11,180,036

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 3]
```diff
@@ -1321,11 +1321,10 @@ void XdsLb::ProcessChannelArgsLocked(const grpc_channel_args& args) {
 
 void XdsLb::UpdateLocked(const grpc_channel_args& args) {
   ProcessChannelArgsLocked(args);
-  // If fallback is configured and the RR policy already exists, update
-  // it with the new fallback addresses.
-  if (lb_fallback_timeout_ms_ > 0 && rr_policy_ != nullptr) {
-    CreateOrUpdateRoundRobinPolicyLocked();
-  }
+  // Note: We have disable fallback mode in the code, so we dont need to update
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Nit: disable -> disabled. dont-> don't. "update the policy" is ambiguous here; suggest saying that ", so we don't need to handle the fallback address changes."

**Comment by AspirinSJL** [comment]
Please squash the commits, thanks!

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc (hunk 3 above)
Comment (vishalpowar) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: done

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.0%    -136 [None]                                                                               -3.55Ki  -0.0%
  -3.7%    -592 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                -592  -3.7%
      [DEL] -1.83Ki grpc_core::(anonymous namespace)::XdsLb::CreateOrUpdateRoundRobinPolicyLocked() [clo -1.83Ki  [DEL]
     -38.6%     -80 grpc_core::(anonymous namespace)::XdsLb::OnFallbackTimerLocked                           -80 -38.6%
     -19.1%     -48 grpc_core::(anonymous namespace)::XdsLb::UpdateLocked                                    -48 -19.1%
      -2.8%     -11 [Unmapped]                                                                               -11  -2.8%

  -0.0%    -728 TOTAL                                                                                -4.12Ki  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.0%    -136 [None]                                                                               -3.55Ki  -0.0%
  -3.7%    -592 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                -592  -3.7%
      [DEL] -1.83Ki grpc_core::(anonymous namespace)::XdsLb::CreateOrUpdateRoundRobinPolicyLocked() [clo -1.83Ki  [DEL]
     -38.6%     -80 grpc_core::(anonymous namespace)::XdsLb::OnFallbackTimerLocked                           -80 -38.6%
     -19.1%     -48 grpc_core::(anonymous namespace)::XdsLb::UpdateLocked                                    -48 -19.1%
      -2.8%     -11 [Unmapped]                                                                               -11  -2.8%

  -0.0%    -728 TOTAL                                                                                -4.12Ki  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by AspirinSJL** [comment]
Please remember to squash the commits before merging. Thanks!

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,040      Total (<)      2,019,390

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,734      Total (<)     11,180,035

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,040      Total (<)      2,019,390

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,732      Total (<)     11,180,028

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Pull request merged**